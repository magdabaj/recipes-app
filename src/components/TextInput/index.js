/**
 *
 * TextInput
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import '../../containers/App/index.css';

function TextInput({ label, name, value, placeholder, onChange, error, type }) {
    let wrapperClass = 'input';
    if (error && error.length > 0) {
        wrapperClass += ' input-error';
    }
    return (
        <div className={'input-data'}>
            <label htmlFor={name} className={'login-label'}>
                {label}
            </label>
            <div className={'field'}>
                <input
                    type={type}
                    name={name}
                    value={value}
                    placeholder={placeholder}
                    onChange={onChange}
                    className={wrapperClass}
                />
            </div>
            {error && <div className={'error-message'}>{error}</div>}
        </div>
    );
}

TextInput.propTypes = {
    label: PropTypes.string,
    name: PropTypes.string,
    onChange: PropTypes.func,
    error: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string,
};

export default TextInput;
