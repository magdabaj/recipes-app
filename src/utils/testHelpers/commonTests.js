const commonTests = (render, useFirstChildAsSnapshot = true) => {
    it('does not log errors in console', () => {
        const spy = jest.spyOn(global.console, 'error')
        render()
        expect(spy).not.toHaveBeenCalled()
    })

    it('should render and match the snapshot',  () => {
        const { container } = render()
        expect(container).not.toBeNull()
        if(useFirstChildAsSnapshot) expect(container.firstChild).not.toBeNull()
        expect(
            useFirstChildAsSnapshot ? container.firstChild : container,
        ).toMatchSnapshot()
    });
}

export default commonTests;