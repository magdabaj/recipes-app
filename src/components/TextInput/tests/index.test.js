import {render, screen} from "@testing-library/react";
import React from "react";
import commonTests from "../../../utils/testHelpers/commonTests";
import TextInput from "../index";

const onChangeMocked = jest.fn()

const renderComponent = () =>
    render(<TextInput
        label={'Email'}
        name={'email'}
        placeholder={'Email'}
        onChange={onChangeMocked}
        error={''}
        value={'test'}
        type={'text'}
    />)

test('renders and shows no errors', () => {
    render(<TextInput
        label={'Email'}
        name={'email'}
        placeholder={'Email'}
        onChange={onChangeMocked}
        error={''}
        value={'test'}
        type={'text'}
    />)
    const inputNode = screen.getByLabelText('Email')
    expect(inputNode.value).toBe('test')
})

test('displays error message, when received a parameter', () => {
    const { getByTestId, getByLabelText } = render(<TextInput
        label={'Email'}
        name={'email'}
        placeholder={'Email'}
        onChange={onChangeMocked}
        error={'some error'}
        value={''}
        type={'text'}
    />)
    const errorMessage = getByTestId('error-message')
    const input = getByLabelText('Email')
    expect(errorMessage).toBeInTheDocument()
    expect(errorMessage.innerHTML).toBe('some error')
    expect(input).toHaveClass('input input-error')
})

commonTests(renderComponent)