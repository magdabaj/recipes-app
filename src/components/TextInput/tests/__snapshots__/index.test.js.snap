// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
<div
  class="input-data"
>
  <label
    class="login-label"
    for="email"
  >
    Email
  </label>
  <div
    class="field"
  >
    <input
      class="input"
      id="email"
      name="email"
      placeholder="Email"
      type="text"
      value="test"
    />
  </div>
  
</div>
`;
